frontend/src/common/SortedCollection.ts

```diff
@@ -131,9 +143,11 @@ export class SortedCollection<Item extends { id: number }> {
 
     this.map[item.id] = item;
 
-    sortedInsert(item, this.list, this.compare);
+    const index = sortedInsert(item, this.list, this.compare);
 
-    this.changeRef += 1;
+    if (index < this.focus.end) {
```

Comment:
Maybe I am missing something but why not `if (index < this.focus.end && index >= this.focus.start)`?

Reply:
Removing or adding a node above the focus will shift the entire focus window up or down. It's only the changes below the viewport that have no effect.